ANSWERS.md

```diff
@@ -1,7 +1,13 @@
 <!-- Answers to the Short Answer Essay Questions go here -->
 
-1. In Mocha, what are the differences between `before`, `after`, `beforeEach`, and `afterEach`? When do they run? What are they used for?
+1.  In Mocha, what are the differences between `before`, `after`, `beforeEach`, and `afterEach`? When do they run? What are they used for?
 
-2. What is the point of Test Driven Development? What do you personally think about this approach?
+* before runs once before the test main test suite runs, after runs once after the main test suite it done. BeforeEach runs before each test suite and afterEach runs after each test suite. They are used set up and close a test db connection, as well as to provide some mock data in our testing db.
 
-3. What is a `spy` in `sinon`? How do we use it to effectively test a `callback`?
+2.  What is the point of Test Driven Development? What do you personally think about this approach?
+
+* The point of TDD is to increase the odds of developing well designed and bug free code, by first thinking and writing test suites that the codebase will be run against.
+

```

Comment:
Very good answer to this question.  Test-driven development forces you to think through the structure of your application before you ever get around to writing anything, and it keeps you continouously thinking about building upon and improving your application.  

**What do you personally think of this approach?**
Personally, I think it rocks!